### PR TITLE
Feat/favicon

### LINKED ISSUE #168: [request]: As a user I want to see a favicon
## 🚀 Feature Proposal

We do not currently have one and this is sad.

### PR DESCRIPTION
**Pull request by elenamarinaki**
### Why

Partially resolves #168 

### What

- copied the favicon package into the `public` folder of the application
- linked the metadata in the `_document.tsx` file

### Checklist

- [ ] Ready to be merged

### CODE REVIEW

File: packages/web/public/favicon/site.webmanifest [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "",
```
Comment (joshuaellis) [inline packages/web/public/favicon/site.webmanifest:2]: This some of this content needs adding, I'd ask @willcompanion or @mylespalmer for the theme color

File: packages/web/src/pages/_document.tsx [hunk 2]
```diff
@@ -40,16 +40,26 @@ export default class Doc extends Document {
...
+          <link rel="manifest" href={'favicon/site.webmanifest'} />
+          <link rel="shortcut icon" href={'favicon/favicon.ico'} />
+          <link rel="apple-touch-icon" href={'favicon/apple-touch-icon.png'} />
+          <link
+            rel="android-chrome-192x192"
+            href={'android-chrome-192x192.png'}
+          />
+          <link
+            rel="android-chrome-256x256"
+            href={'android-chrome-256x256.png'}
```
Comment (joshuaellis) [inline packages/web/src/pages/_document.tsx:60]: this has an incorrect path?

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/web/src/client/components/Site/SiteLayout.tsx [hunk 3]
```diff
@@ -34,6 +36,44 @@ export const Layout = ({
 
   const [showSplash, setShowSplash] = useState(true)
 
+  // :: ------------------------------------------------------
+  let counter = 0
+
+  setInterval(() => {
```
Comment (joshuaellis) [inline packages/web/src/client/components/Site/SiteLayout.tsx:42]: Here's an article that will help you understand how to use 'setInterval' in react - https://sebhastian.com/setinterval-react/

File: packages/web/src/client/components/Site/SiteLayout.tsx [hunk 4]
```diff
@@ -34,6 +36,44 @@ export const Layout = ({
...
+    if (typeof window !== 'undefined') {
+      if (document.querySelector("link[rel='icon']") !== null)
+        document.querySelector("link[rel='icon']")?.remove()
+
+      document
+        .querySelector('Head')
+        ?.insertAdjacentHTML(
+          'beforeend',
+          '<link rel="icon" href="' +
```
Comment (joshuaellis) [inline packages/web/src/client/components/Site/SiteLayout.tsx:51]: Im not the biggest fan of this method, it feels expensive to be destroying and creating nodes on the fly. Can we not just change the source of the link?

We could attach and ID to it, for example.

**Review by joshuaellis — CHANGES_REQUESTED** [review]

**Comment by elenamarinaki** [comment]
@joshuaellis 
in this last PR, I am bringing back the `useEffect` to run the timer, since this has to be done client-side.

I am getting the `id` of the `setInterval` callback, so that I can clear it at the end. Inside, I am running the actual code:
1. I am checking if there is a `document` (it should be, we're on the client)
2. selecting the `<link />` element
3. I am mutating its `href` with the `setAttribute` method
4. lastly, I am updating the counter correctly, with a callback on its previous state

---

This is still not running, so I am still investigating as to why 🤔

File: packages/web/src/client/components/Site/SiteLayout.tsx [hunk 5]
```diff
@@ -34,6 +36,59 @@ export const Layout = ({
...
+     * Animate logo function, could be doing in a raf call?
+     */
+    const animateLogo = () => {
+      if (link) {
+        link.setAttribute(
+          'href',
+          `/images/favicon-sequence/Companion_Favicon${counter}.png`
+        )
+
+        if (counter === 150) {
```
Comment (joshuaellis) [inline packages/web/src/client/components/Site/SiteLayout.tsx:72]:
```suggestion
        if (counter === 149) {
```

**Review by joshuaellis — CHANGES_REQUESTED** [review]

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**